llama_poe/poe_api/server.py

```diff
@@ -0,0 +1,121 @@
+import copy
+import logging
+import os
+from typing import Any, Dict
+
+import uvicorn.config
+from fastapi import Depends, FastAPI, HTTPException, Request, Response
+from fastapi.exceptions import RequestValidationError
+from fastapi.responses import HTMLResponse
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+from poe_api import llama_handler
+from poe_api.types import AddDocumentsRequest
+from poe_api.utils import LoggingMiddleware
+from sse_starlette.sse import EventSourceResponse
+
+from fastapi_poe.types import (
+    QueryRequest,
+    ReportErrorRequest,
+    ReportFeedbackRequest,
+    SettingsRequest,
+)
+
+global logger
+logger = logging.getLogger("uvicorn.default")
+
+http_bearer = HTTPBearer()
+BEARER_TOKEN = os.environ.get("POE_API_KEY")
+assert BEARER_TOKEN is not None
+
+
+def exception_handler(request: Request, ex: HTTPException):
+    del request
+    logger.error(ex)
+
+
+def auth_user(
+    authorization: HTTPAuthorizationCredentials = Depends(http_bearer),
+) -> None:
+    if authorization.scheme != "Bearer" or authorization.credentials != BEARER_TOKEN:
+        raise HTTPException(
+            status_code=401,
+            detail="Invalid API key",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+
+
+app = FastAPI()
+app.add_exception_handler(RequestValidationError, exception_handler)
+
+# Uncomment this line to print out request and response
+app.add_middleware(LoggingMiddleware)
+logger.info("Starting")
+
+log_config = copy.deepcopy(uvicorn.config.LOGGING_CONFIG)
+log_config["formatters"]["default"]["fmt"] = "%(asctime)s - %(levelname)s - %(message)s"
+
+
+@app.get("/")
+async def index() -> Response:
+    url = "https://poe.com/create_bot?api=1"
+    return HTMLResponse(
+        "<html><body><h1>FastAPI Poe bot server</h1><p>Congratulations! Your server"
```

Comment:
```suggestion
        "<html><body><h1>LlamaIndex Poe bot server</h1><p>Congratulations! Your server"
```
Differentiate it from the more basic template we're already providing

Reply:
Thanks!